dht/i_partitioner.hh

```diff
@@ -848,13 +848,8 @@ template<>
 struct hash<dht::token> {
     size_t operator()(const dht::token& t) const {
         const auto& b = t._data;
-        if (b.size() == sizeof(size_t)) { // practically always
-            return read_le<size_t>(reinterpret_cast<const char*>(b.data()));
-        }
-        return hash_large_token(b);
+        return read_le<size_t>(reinterpret_cast<const char*>(b.data()));
```

Comment:
This breaks on a machine where sizeof(size_t) != 8. Can be fixed by s/size_t/uint64_t/. The fact that the function returns size_t doesn't hurt.

Reply:
done